text/0086-sentry-bundler-plugins-api.md

```diff
@@ -0,0 +1,103 @@
+- Start Date: 2023-04-13
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/86
+- RFC Status: draft
+
+# Summary
+
+This RFC intends to discuss the future API of the [Sentry JavaScript Bundler Plugins](https://github.com/getsentry/sentry-javascript-bundler-plugins).
+
+# Motivation
+
+With the major bump to v2 of the bunlder plugins, we have the opportunity to rethink the existing API that was orignially built on top of the [Sentry Webpack Plugin](https://github.com/getsentry/sentry-webpack-plugin).
+
+Facts to consider when deciding on an API:
+
+- The bump to v2 of unplugin will imply a major bump of the Sentry Webpack plugin and its switch to our unplugin based architecture.
+- We recently introduced debug IDs as a source map resolving mechanism. This results in different needed options needed when running the plugins. Some options must be added, some removed.
+- Releases are not a required parameter anymore for source maps upload.
+
+As an additional thought: The new API should be seen as a "clean-slate" API - it's less about what has been and more about what the optimal API should look like.
+
+# Supporting Data
+
+- Debug ID RFC: https://github.com/getsentry/rfcs/blob/ab0f75567d95e247b5cc3f09c466b7254844e654/text/0081-sourcemap-debugid.md
+- Debug ID changes related to the unplugin based Sentry bundler plugins: https://github.com/getsentry/sentry-javascript-bundler-plugins/issues/191
+
+# Options Considered
+
+The API generally only defines the options passed to the individual plugins. The options are the same for all bundlers.
+
+For the sake of simplicity of this RFC, we will not decide on the JSdoc. All JSdoc below is just to illustrate what the options are used for.
+
+Proposed API (TypeScript typedef):
+
+```ts
+interface Options {
+  org: string; // Previously this was optional and could have been set with `env.SENTRY_ORG` or `.sentryclirc` - for the sake of explicity, we remove these configuration methods
```

Comment:
I understand how this simplifies a lot, but a little concerned with removing the environmental variables config.

I'd like to keep it because I feel that it reduces the changes needed for users CI/CD setups, and also we don't need to update things like the Vercel integration.

I'm all for `sentryclirc` - bundler plugin users shouldn't have to be burdened with the fact that we use `sentry-cli` under the hood.

Reply:
> I understand how this simplifies a lot, but a little concerned with removing the environmental variables config.
> 
> I'd like to keep it because I feel that it reduces the changes needed for users' CI/CD setups, and also we don't need to update things like the Vercel integration.

The way I thought about solving this particular issue would just be to have a very explicit `org: process.env.ORG_SLUG` in the docs/readmes/wizard snippets.

Something bout having env vars feed into bundler plugins feels odd... I get the convenience aspect though.

Do you think it would be an option for us to just unship env vars for now, and reintroduce them later if people see good use for them?

Reply:
> Do you think it would be an option for us to just unship env vars for now, and reintroduce them later if people see good use for them?

My opinion is just to keep environmental variables, it's an acceptable amount of complexity in my eyes, esp for folks who have their CI pipelines set up that way. Just my opinion though, so I'm fine to go with what majority wants.

Reply:
Ok let's keep them then!